src/app/components/search/useSearch.ts

```diff
@@ -43,9 +47,16 @@ export default function useSearch() {
   });
 
   useDebounce({
-    callback: () => handleSearch(departmentCode ?? "", number, ge ?? ""),
+    callback: () =>
+      handleSearch(
+        departmentCode ?? "",
+        number,
+        ge ?? "",
+        numberRange,
+        creditRange,
+      ),
     delay: 100,
-    dependencies: [departmentCode, number, ge],
+    dependencies: [departmentCode, number, ge, numberRange, creditRange],
```

Comment:
Spread numberRange and creditRange like so: `...numberRange, ...creditRange`, to avoid triggering this effect unnecessarily